### ISSUE TITLE
Creating Launch Templates to Create, Persist, Bundle, and Delete Configurations

### ISSUE THREAD

**Issue by xee5ch** [avishayil/secure_ec2#5]
Thanks again for this cool tool. I have another feature request, per our convo on social media last night/this morning re launch templates.

Why would I also suggest this? I worked with a lot of devs who don't enjoy AWS that much, and [re tyranny of the default](https://en.wikipedia.org/wiki/Secure_by_default), will pick the easiest settings to run a new instance in the _web console_, on their first day or nth day on a project. If this tool leaves the configs for those devs _in the web console as well_ and preseeds them, this tool has a very nice multiplier effect. :-)

Most, if not all, of the settings you are setting per invocation could be optionally saved (or optionally not saved) to a immutable launch template. They also can increment versions. Tagging those launch templates means you can also group and clean them as needed. I think it would be great to not have to memorize settings and just say `secure_ec2 config --save` (just spitballing here) and persist a new or updated version of an LT. You can subsequently review and run 1 or n>1 instances with that LT, and maybe even delete them when done.

[The docs for boto3 `run_instances()` function](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Client.run_instances) suggest this is definitely doable and supported.

Like my previous comment/suggestion, I am willing to help implement this with/for you on GitHub if you would like. Let me know!

**Comment by avishayil** [comment]
Thank you very much @xee5ch !
I'm starting to think that it might be a good idea to use `secure_ec2 config` for the set up configuration that would provision the launch template, but still allowing non-wizard scenarios to save the configuration via `secure_ec2 --save`.
WDYT?

**Comment by xee5ch** [comment]
> Thank you very much @xee5ch ! I'm starting to think that it might be a good idea to use `secure_ec2 config` for the set up configuration that would provision the launch template, but still allowing non-wizard scenarios to save the configuration via `secure_ec2 --save`. WDYT?

Sounds good. Might give it a whack, if not this weekend, the next!

**Comment by avishayil** [comment]
I think we're going to need a refactor here for the `config` command. I have something on my mind for implementing it.

**Comment by xee5ch** [comment]
So you are suggesting I hold off then, @avishayil?

**Comment by avishayil** [comment]
@xee5ch I think os, because there is some massive refactor that i'm planning to complete, and it will change the way that tool is invoked.
You can review the progress here (most of the functionality and tests are in place, I need to complete the documentation and some GIFs): https://github.com/avishayil/secure_ec2/tree/feat/refactor
TLDR:
```
secure_ec2 config
secure_ec2 launch
```

**Comment by avishayil** [comment]
solved with #8
